### PR TITLE
Security doc fix and improved warning message

### PR DESCRIPTION
**Pull request by galderz**
https://github.com/quarkusio/quarkus/issues/9664

### CODE REVIEW

File: core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigDiagnostic.java [hunk 1]
```diff
@@ -48,7 +48,7 @@ public static void deprecated(String name) {
     }
 
     public static void unknown(String name) {
-        log.warnf("Unrecognized configuration key \"%s\" was provided; it will be ignored", name);
+        log.warnf("Unrecognized configuration key \"%s\" was provided; it will be ignored; verify that the dependency extension for this configuration is set", name);
```
Comment (emmanuelbernard) [inline core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigDiagnostic.java:51]: if you display this, you also should add "or that you did not make a typo".

**Comment by galderz** [comment]
@emmanuelbernard Pushed a fix to improve the message.

**Review by sberyozkin — APPROVED** [review]
@galderz IMHO it looks good. Small proposal, drop a 2nd `that` in the message as it has become a bit long for it to become:
`...verify that the dependency extension for this configuration is set or you did not make a typo`

But it is just a very small thing, cc @gsmet @emmanuelbernard please merge if you are happy with the PR

**Comment by galderz** [comment]
@sberyozkin Done

**Comment by sberyozkin** [comment]
@galderz Thanks :-), looks like everyone is happy with your fix so let me merge, cheers

**Pull request merged**